lib/datadog/statsd/connection_cfg.rb

```diff
@@ -10,28 +10,31 @@ class ConnectionCfg
       DEFAULT_PORT = 8125
 
       def initialize(host: nil, port: nil, socket_path: nil)
+        initialize_with_constructor_args(host, port, socket_path) || initialize_with_env_vars || initialize_with_defaults
+      end
 
-        # Try with constructor args
-        if setup_with(host, port, socket_path)
-          return
-        end
+      def initialize_with_constructor_args(host, port, socket_path)
+        try_initialize_with(host, port, socket_path,
+          "Both host/port and socket_path constructor arguments are set.  Set only one or the other.",
```

Comment:
Minor: Perhaps it would be helpful to include their values to help debugging :) (Same suggestion applies below as well)